sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -787,7 +789,7 @@ case class HashAggregateExec(
          |    $unsafeRowKeys, ${hashEval.value});
          |  if ($unsafeRowBuffer == null) {
          |    // failed to allocate the first page
-         |    throw new OutOfMemoryError("No enough memory for aggregation");
+         |    throw new $oomeClassName("No enough memory for aggregation");
```

Comment:
Hi, @ueshin . Is this the final place? If not, can we have a separate JIRA issue for this?

Reply:
Yes, I think so based on my investigation. I grep-ed with "OutOfMemoryError" and checked the suspicious places.